### DISCUSSION TITLE
Socket Timeout

### DISCUSSION THREAD

**Discussion by jbenghiat** [homebridge-simplisafe3/homebridge-simplisafe3#173] [Q&A]
I have the integration working -- the plug-in finds and creates sensors, however, I'm not able to establish a socket.io connection. Are there any known issues here, or something that needs to change in the configuration?

`SSAPI socket connect_error: timeout`
`socket.io-client:manager readyState closed`
`socket.io-client:manager opening https://api.simplisafe.com/v1/user/XXXXXX`
`engine.io-client:socket creating transport "websocket"`
`engine.io-client:socket setting transport websocket`
`socket.io-client:manager connect attempt will timeout after 20000`
`socket.io-client:manager connect attempt timed out after 20000`
`engine.io-client:socket socket close with reason: "forced close"`
`engine.io-client:socket socket closing - telling transport to close`
`socket.io-client:manager connect_error`
`socket.io-client:manager cleanup`

**Comment by shamoon** [comment]
Youre the second person today who might be having this issue but Im not seeing it. Did you verify that you are able to login to the webapp? Also did you get a "two-factor" email when starting that you successfully clicked?

**Reply by jbenghiat** [reply]
I figured out the issue -- looking at the SS webapp, it includes socket.io 1.5.1. I changed the dependency to 1.5.1, and success. 1.7.4 is working as well.

**Reply by shamoon** [reply]
I get the feeling you were talking about the Home-assistant integration, this plugin is for Homebridge

**Reply by jbenghiat** [reply]
I’m using Homebridge — what’s giving you the impression I’m using Home-assistant?

**Reply by shamoon** [reply]
Dunno, logs looked different, this plugin socket.io is at 2.3.0 ? https://github.com/homebridge-simplisafe3/homebridge-simplisafe3/blob/cb444f1c1f8e11b27b96cbb9f8fb32f2ebdafcf3/package-lock.json#L6490

**Reply by shamoon** [reply]
Oh or are you suggesting we downgrade?

**Reply by jbenghiat** [reply]
Thanks — I turned on the node.js debug flag, so there’s some extra messages. 

Yes — downgrading the dependency is what worked for me. SS seems to be running v1 of socket.io on their end. I’m not sure if there is another workaround, but that did the trick for me.

**Reply by shamoon** [reply]
Weird that I’m not having any issues and doesn’t seem it’s widespread (yet?) either. Can I ask what OS and version of node-js you’re using?

**Reply by jbenghiat** [reply]
I'm running on Mac OS 10.15 -- my plan was to get all my devices set up on and tested the Mac before moving over to a pi.

Homebridge 1.1.7
Node.js 15.5.0
Npm 7.3.0

**Reply by shamoon** [reply]
Interesting & odd. Curious to see if you get this on the Pi. 
I test on macOS and run my instance on Ubuntu and neither are showing this issue.

One thing to note, that node version isnt actually supported by Homebridge: https://github.com/homebridge/homebridge/wiki/How-To-Update-Node.js 
If you feel up to it maybe try 14.x (which is what Im on)?

**Reply by ronnyf** [reply]
> I figured out the issue -- looking at the SS webapp, it includes socket.io 1.5.1. I changed the dependency to 1.5.1, and success. 1.7.4 is working as well.

sorry for a dumb sounding question but how did you change that dependency ?

**Reply by jbenghiat** [reply]
> > I figured out the issue -- looking at the SS webapp, it includes socket.io 1.5.1. I changed the dependency to 1.5.1, and success. 1.7.4 is working as well.
> 
> sorry for a dumb sounding question but how did you change that dependency ?

In your homebridge-simplisafe3 installation, edit the package.json file, and change the version number in the dependencies section. 
In a terminal window, make that the active directory. 
Run npm install

**Reply by ronnyf** [reply]
thanks a lot, this worked (obviously) super well ...
